functions.php

```diff
@@ -374,3 +374,138 @@ function wp_get_list_item_separator() {
     return __( ', ' );
   }
 }
+
+
+// Update Registration form and functions
+add_action('login_head', function () {
+  ?>
+      <style>
+          #registerform>p:first-child {
+              display: none;
+          }
+      </style>
+  
+      <script type="text/javascript" src="<?php echo site_url('/wp-includes/js/jquery/jquery.js'); ?>"></script>
+      <script type="text/javascript">
+          jQuery(document).ready(function($) {
+              $('#registerform > p:first-child').css('display', 'none');
+          });
+      </script>
+  <?php
+  });
+  
+  //Remove error for username, only show error for email only.
+  add_filter('registration_errors', function ($wp_error, $sanitized_user_login, $user_email) {
+      if (isset($wp_error->errors['empty_username'])) {
+          unset($wp_error->errors['empty_username']);
+      }
+  
+      if (isset($wp_error->errors['username_exists'])) {
+          unset($wp_error->errors['username_exists']);
+      }
+      return $wp_error;
+  }, 10, 3);
+  
+  
+  add_action('login_form_register', function () {
+      if (isset($_POST['user_login']) && isset($_POST['user_email']) && !empty($_POST['user_email'])) {
+          $_POST['user_login'] = $_POST['user_email'];
+      }
+  });
+  
+  
+  // Add custom field to the registration form
+  add_action('register_form', function () {
+  ?>
+  
+      <p class="user-password-wrap">
+          <label for="password"><?php _e('Password'); ?></label>
+          <input type="password" name="password" id="password" class="input password-input" size="20" value="" autocomplete="off" />
```

Comment:
additionally, I don't recommend using Id's like this. id "password" may not be unique, which would make html non-compliant.